gender_analysis/common.py

```diff
@@ -111,124 +112,50 @@
 # TODO(elsa): Investigate doctest errors in this file, may be a result of
 # my own system, not actual code errors
 
+def load_csv_to_dict(file_path):
+    """
+    Loads a csv file
+
+    :param file_path: can be a string or Path object
+    :return: a list of strings
+    """
+    if isinstance(file_path, str):
+        file_path = Path(file_path)
 
-class FileLoaderMixin:
-    """ The FileLoaderMixin loads files either locally or
-    remotely from Github (if run from an ipython notebook)
+    file_type = file_path.parts[-1][file_path.parts[-1].rfind('.'):]
+    file = open(file_path, encoding='utf-8')
```

Comment:
This file handle never gets closed. What you want to do is

0: check the file type, throw exception if it's the wrong type
1. open the file handle and call read() or readlines(), saving the result to a variable
2. close the file handle
3. return the variable you saved everything to